File: frontend/src/types/timeline.ts
export interface TimelineEvent {
  id: string;
  type: 'surgery' | 'imaging' | 'vitals' | 'emergency_room';
  timestamp: string;
  patientId: number;
  data: {
    surgeonName?: string;
    procedure?: string;
    attendingPhysician?: string;
    chiefComplaint?: string;
    modality?: string;
    radiologistNote?: string;
    bpm?: number;
    bp?: string;
  };
  flag?: 'post-discharge';
  source: 'registry' | 'pacs' | 'vitals';
}

export interface GroupedTimelineEvent extends TimelineEvent {
  start?: string;
  end?: string;
  children?: TimelineEvent[];
}

export interface TimelineResponse {
  parents: GroupedTimelineEvent[];
  standalone: TimelineEvent[];
  partial: boolean;
  warning?: string;
}

export type UserRole = 'doctor' | 'nurse' | 'intern';

File: backend/models/timeline.py
from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel

from models.enums.data_source import DataSource
from models.enums.event_type import EventType


class TimelineEvent(BaseModel):
    id: str
    type: EventType
    source: DataSource
    timestamp: datetime
    patientId: int
    data: dict
    start: Optional[datetime] = None
    end: Optional[datetime] = None
    children: List["TimelineEvent"] = []
